### DISCUSSION TITLE
Changing TRILIUM_DATA_DIR - `export` doesn't do the trick

### DISCUSSION THREAD

**Discussion by kjgcoop** [TriliumNext/Trilium#6069] [Ideas] [answered]
I read in Classic Trilium's FAQ (https://github.com/zadam/trilium/wiki/FAQ) that the data dir can be changed like so:
`TRILIUM_DATA_DIR=/home/me/path/to/data/dir TRILIUM_PORT=12345 trilium `

As TriliumNext seems to be started by `npm run start-server`, I ran a variant of that: 
`TRILIUM_DATA_DIR=/home/me/trilium-data TRILIUM_PORT=8000 npm run start-server`

It's still starting with the message:
`> cross-env TRILIUM_SAFE_MODE=1 TRILIUM_DATA_DIR=./data TRILIUM_ENV=dev TRILIUM_SYNC_SERVER_HOST=http://tsyncserver:4000 nodemon src/www.ts`

I let it start all the way to see if maybe the message was inaccurate, but it appears it really is using ./data. I also tried putting a new value for TRILIUM_DATA_DIR in .bashrc, to no avail.

Just to see what would happen, I tried setting a value `triliumDataDir` this in data/config.ini, but it ignored it. 

Any help would be much appreciated. Thank you!

**Comment by capi** [comment, marked as answer]
Hi, looking at `package.json`, the `npm run start-server` is defined as
```
"start-server": "cross-env TRILIUM_SAFE_MODE=1 TRILIUM_DATA_DIR=./data TRILIUM_ENV=dev TRILIUM_SYNC_SERVER_HOST=http://tsyncserver:4000 nodemon src/www.ts"
```
which means that the environment variables are overwritten.

The documented TRILIUM_DATA_DIR and TRILIUM_PORT are working on the compiled electron apps or the docker containers, where the launcher does not overwrite the information. I use it that way personally and it works pretty well.

If you want to run the application with `npm run start-server`, the easiest solution seems to be to edit package.json to your liking.

**Comment by kjgcoop** [comment]
@capi Many thanks!

For anybody else who has the notion to run both classic and next on the same database but at different ports, don't forget to add `TRILIUM_PORT=8000` (or whatever port you like) to the aforementioned line. Otherwise both instances will read the port from the same config file in data/.
